plugin/PlayAssetDelivery/src/main/java/com/google/play/core/godot/assetpacks/PlayAssetDelivery.java

```diff
@@ -20,15 +20,81 @@
 
 import org.godotengine.godot.Godot;
 import org.godotengine.godot.plugin.GodotPlugin;
+import org.godotengine.godot.plugin.SignalInfo;
 
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+import java.util.HashSet;
+
+/**
+ * This class is served as a middleware, to expose the Play Core Java library to the Godot runtime,
+ * so Godot Play Asset Delivery front-end API is able to access the related Java APIs
+ * This class is intended to be compile to an .AAR Android Library, which can be loaded into
+ * Godot Engine as an Godot Android Plugin
+ */
 public class PlayAssetDelivery extends GodotPlugin {
+
     public PlayAssetDelivery(Godot godot) {
         super(godot);
     }
 
     @NonNull
     @Override
     public String getPluginName() {
-        throw new UnsupportedOperationException("Method not implemented yet!");
+        return "PlayAssetDelivery";
+    }
+
+    /**
+     *
+     * @return a list of all the available plugin method names that the Godot runtime can call
+     */
+    @NonNull
+    @Override
+    public List<String> getPluginMethods() {
+        return Arrays.asList("cancel", "fetch", "getAssetLocation", "getPackLocation",
+                "getPackLocations", "getPackStates", "removePack", "showCellularDataConfirmation");
+    }
+
+    /**
+     *
+     * @return a set containing all the signals the Godot runtime is able to receive
```

Comment:
nit: For all comments, add periods to indicate the end of a sentence rather than a new line.